packages/mwp-router/src/AsyncRoute.jsx

```diff
@@ -10,90 +10,86 @@ type Props = {
 	location: URL,
 	history: RouterHistory,
 };
-type State = {
-	childRoutes: Array<PlatformRoute>,
-	_routesCache: { [string]: Array<PlatformRoute> },
+type ComponentState = {
+	component: React$ComponentType<*>,
+	_componentCache: { [string]: React$ComponentType<*> },
 };
+type State = ComponentState;
```

Comment:
This looks redundant, but it allows me to write a state-setting function that is explicitly limited to updating 'component state', while the component itself can get other state params in the future.